src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -6962,7 +6968,37 @@ static void sofia_handle_sip_r_invite(switch_core_session_t *session, int status
 				}
 			}
 		}
-
+		
+		// Feature flipper:  We want the chanvar to override the profile setting.
+		if(status == 200 && switch_channel_test_flag(channel, CF_RECOVERED) && !tech_pvt->recovered_call_route_fixed && ((!profile->disable_recovery_record_route_fixup && drrrf_chanvar == SWITCH_FALSE ) || (profile->disable_recovery_record_route_fixup && !drrrf_chanvar_zstr && drrrf_chanvar == SWITCH_FALSE))) {
+			const char *sip_invite_record_route = switch_channel_get_variable(channel, "sip_invite_record_route");
+			const char *total_recovery_rr_fixups = switch_channel_get_variable(channel, "total_recovery_record_route_fixups");
+			
+			if(!zstr(sip_invite_record_route)) {
+				int32_t total_recovery_rr_fixups_int = 0;
+				nta_agent_t *curagent = nua_get_agent(profile->nua);
+				const char *sip_call_id = switch_channel_get_variable(channel, "sip_call_id");
+				su_home_t *home = nua_handle_get_home(nh);
+				sip_record_route_t *rrtemp = sip_record_route_make(home, sip_invite_record_route);
+				nta_leg_t *curleg = nta_leg_by_call_id(curagent, sip_call_id);
+				
+				nta_leg_client_reroute(curleg, rrtemp, sip->sip_contact, 1);
+				
+				if(!zstr(total_recovery_rr_fixups)) {
+					total_recovery_rr_fixups_int = atoi(total_recovery_rr_fixups);
+				}
+				
+				switch_channel_set_variable_printf(channel, "total_recovery_record_route_fixups", "%d", total_recovery_rr_fixups_int);
```

Comment:
total_recovery_rr_fixups_int will always be zero as we are not incrementing it when we set the channel variable

Reply:
Missing increment fixed.